src/clientSideScene/sceneEntities.ts

```diff
@@ -1584,6 +1667,448 @@ export class SceneEntities {
+          execState.variables[variableDeclarationName],
+          variableDeclarationName
+        )
+        if (err(sketch)) return
+        const sgPaths = sketch.paths
+        const orthoFactor = orthoScale(sceneInfra.camControls.camera)
+
+        const varDecIndex = Number(sketchEntryNodePath[1][0])
+
+        this.updateSegment(
+          sketch.start,
+          0,
+          varDecIndex,
+          _ast,
+          orthoFactor,
+          sketch
+        )
+        sgPaths.forEach((seg, index) =>
+          this.updateSegment(seg, index, varDecIndex, _ast, orthoFactor, sketch)
+        )
+      },
+      onClick: async (args) => {
+        const firstProfileIndex = Number(sketchNodePaths[0][1][0])
+        const nodePathWithCorrectedIndexForTruncatedAst = structuredClone(
+          mod.pathToNode
+        )
+
+        nodePathWithCorrectedIndexForTruncatedAst[1][0] =
+          Number(nodePathWithCorrectedIndexForTruncatedAst[1][0]) -
+          firstProfileIndex
+        // If there is a valid camera interaction that matches, do that instead
+        const interaction = sceneInfra.camControls.getInteractionType(
+          args.mouseEvent
+        )
+        if (interaction !== 'none') return
+        // Commit the arc to the full AST/code and return to sketch.idle
+        const mousePoint = args.intersectionPoint?.twoD
+        if (!mousePoint || args.mouseEvent.button !== 0) return
+
+        const _node = getNodeFromPath<VariableDeclaration>(
+          _ast,
+          sketchEntryNodePath || [],
+          'VariableDeclaration'
+        )
+        if (trap(_node)) return
+        const sketchInit = _node.node?.declaration.init
+
+        let modded = structuredClone(_ast)
+        if (sketchInit.type === 'PipeExpression') {
+          // Calculate end angle based on final mouse position
+
+          // Calculate the final 'to' point using the existing radius and the final end angle
+          const finalP3: [number, number] = [mousePoint.x, mousePoint.y]
+
+          const moddedResult = changeSketchArguments(
+            modded,
+            kclManager.variables,
+            {
+              type: 'path',
+              pathToNode: mod.pathToNode,
+            },
+            {
+              type: 'circle-three-point-segment',
+              p1,
+              p2,
+              p3: finalP3,
+            }
+          )
+          if (err(moddedResult)) return
+          modded = moddedResult.modifiedAst
+
+          const newCode = recast(modded)
+          if (err(newCode)) return
+          const pResult = parse(newCode)
+          if (trap(pResult) || !resultIsOk(pResult))
+            return Promise.reject(pResult)
+          _ast = pResult.program
+
+          // Update the primary AST and unequip the arc tool
+          await kclManager.executeAstMock(_ast)
+          sceneInfra.modelingSend({ type: 'Finish arc' })
+          await codeManager.updateEditorWithAstAndWriteToFile(_ast)
```

Comment:
This will not wait to write to file right? Like internally it'll wait for editor but not the file write due to issues we've had in the past?

Reply:
`updateEditorWithAstAndWriteToFile` is awaited a bunch


Just checked, yeah looks like the write-to-file part is not awaited. but the function doesn't actually need to be async 🤦, Don't think I should clean that up in the PR.

Reply:
Roger :saluting_face: